File: 2/node/objective2a.js
function fib1(x){
    if(x <= 0)
        return 0;
    if(x === 1)
        return 1;
    return fib1(x-1) + fib1(x-2);
}

function fib2(x){
    return fib2helper(0, 1, x-1);
}

function fib2helper(x, y, z){
    if(z < 0)
        return 0;
    if(z == 0)
        return y;
    return fib2helper(y, y+x, z-1);
}

functions = [fib1, fib2];

functions.forEach(function(func){
    var start = new Date();
    for(i = 0; i < 40; ++i)
        console.log(func(i));
    var end = new Date();
    var diff = end - start;
    console.log(`${func}: Elapsed time: ${diff}`);
});

File: 2/java/objective2a.java
import java.util.Date;
class objective1 {
    public static void main(String[] args){
        long time = time();
        for(int i = 0; i < 40; ++i)
            System.out.println(fib1(i));
        time = time() - time;
        System.out.println("fib1() elapsed time: " + time);

        time = time();
        for(int i = 0; i < 40; ++i)
            System.out.println(fib2(i));
        time = time() - time;
        System.out.println("fib2() elapsed time: " + time);
    }

    public static long time(){
        Date dt = new Date();
        return dt.getTime();
    }

    public static int fib1(int x){
        if(x <= 0)
            return 0;
        if(x == 1)
            return 1;
        return fib1(x-1) + fib1(x-2);
    }

    public static int fib2(int z){
        return fib2helper(0, 1, z-1);
    }

    public static int fib2helper(int x, int y, int z){
        if(z < 0)
            return 0;
        if(z == 0)
            return y;
        return fib2helper(y, x+y, z-1);
    }
}
